miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditServiceRecordController.java

```diff
@@ -0,0 +1,147 @@
+import uk.ac.bbsrc.tgac.miso.core.data.SequencerReference;
+import uk.ac.bbsrc.tgac.miso.core.data.SequencerServiceRecord;
+import uk.ac.bbsrc.tgac.miso.core.factory.DataObjectFactory;
+import uk.ac.bbsrc.tgac.miso.core.manager.FilesManager;
+import uk.ac.bbsrc.tgac.miso.core.manager.RequestManager;
+
+@Controller
+@RequestMapping("/stats/sequencer/servicerecord")
+@SessionAttributes("serviceRecord")
+public class EditServiceRecordController {
+  
+  protected static final Logger log = LoggerFactory.getLogger(EditServiceRecordController.class);
+  
+  @Autowired
+  private RequestManager requestManager;
+
+  @Autowired
+  private FilesManager filesManager;
+
+  @Autowired
+  private DataObjectFactory dataObjectFactory;
+
+  public void setRequestManager(RequestManager requestManager) {
+    this.requestManager = requestManager;
+  }
+
+  public void setFilesManager(FilesManager filesManager) {
+    this.filesManager = filesManager;
+  }
+
+  public void setDataObjectFactory(DataObjectFactory dataObjectFactory) {
+    this.dataObjectFactory = dataObjectFactory;
+  }
+  
+  public Map<Integer, String> populateServiceRecordFiles(SequencerServiceRecord record) throws IOException {
+    if (record.getId() != AbstractSequencerServiceRecord.UNSAVED_ID) {
+      Map<Integer, String> fileMap = new HashMap<Integer, String>();
+      for (String s : filesManager.getFileNames(SequencerServiceRecord.class, String.valueOf(record.getId()))) {
+        fileMap.put(s.hashCode(), s);
+      }
+      return fileMap;
+    }
+    return Collections.emptyMap();
+  }
+  
+  @ModelAttribute("maxLengths")
+  public Map<String, Integer> maxLengths() throws IOException {
+    return requestManager.getServiceRecordColumnSizes();
+  }
+
+  @RequestMapping(value = "/{recordId}", method = RequestMethod.GET)
+  public ModelAndView viewServiceRecord(@PathVariable(value = "recordId") Long recordId, ModelMap model) throws IOException {
+    SequencerServiceRecord sr = requestManager.getSequencerServiceRecordById(recordId);
+    if (sr != null) {
+      model.put("serviceRecord", sr);
+      model.put("serviceRecordFiles", populateServiceRecordFiles(sr));
+    } else {
+      throw new IOException("Cannot retrieve the requested Service Record");
+    }
+    return new ModelAndView("/pages/editServiceRecord.jsp", model);
+  }
+  
+  @RequestMapping(value = "/new/{sequencerReferenceId}", method = RequestMethod.GET)
+  public ModelAndView newServiceRecord(@PathVariable(value = "sequencerReferenceId") Long sequencerReferenceId, ModelMap model) throws IOException {
+    SequencerReference sequencer = requestManager.getSequencerReferenceById(sequencerReferenceId);
+    if (sequencer == null) {
+      throw new IOException("Cannot retrieve the requested Sequencer Reference");
+    }
+    SequencerServiceRecord record = dataObjectFactory.getSequencerServiceRecord();
+    record.setSequencerReference(sequencer);
+    model.put("serviceRecord", record);
```

Comment:
adding ModelKeys enums for ServiceRecord, SequencerReference, and Stats controllers